### PR TITLE
Update Daisy with the overridden APIs endpoints

### PR DESCRIPTION
**Pull request by MahmoudNada0**
Update Daisy with the overridden APIs endpoints by passing the updated clients to the workflow

/cc @darthmelonder
/assign @darthmelonder

### CODE REVIEW

File: cli_tools/common/image/importer/daisy_inflater.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func newDaisyInflater(request ImageImportRequest, fileMetadata imagefile.Metadat
 
 	workflowProvider := func() (*daisy.Workflow, error) {
 		wf, err := daisyutils.ParseWorkflow(path.Join(request.WorkflowDir, wfPath), vars,
-			request.Project, request.Zone, request.ScratchBucketGcsPath, request.Oauth, request.Timeout.String(), request.ComputeEndpoint,
+			request.Project, request.Zone, request.ScratchBucketGcsPath, request.Oauth, request.Timeout.String(), request.EndpointsOverride,
```
[the remark this answers is not in the document]
Comment (MahmoudNada0) [reply cli_tools/common/image/importer/daisy_inflater.go:109]: Yes you're right, the updated compute & storage clients are being already passed to the NewInflater function.

**Review by darthmelonder — COMMENTED** [review]
I think we are not considering the override for inspection which might fail in TPC: https://github.com/GoogleCloudPlatform/compute-image-import/blob/main/cli_tools/common/disk/inspect.go#L51
Let me know your thoughts?

**Comment by MahmoudNada0** [comment]
> > I think we are not considering the override for inspection which might fail in TPC: https://github.com/GoogleCloudPlatform/compute-image-import/blob/main/cli_tools/common/disk/inspect.go#L51 Let me know your thoughts?
> 
> Also, I think the best way here might be: to add pre-run hooks to the workflow worker below, based on environment variables:
> 
> https://github.com/GoogleCloudPlatform/compute-image-import/blob/7f68453cd337a954156cbe27eb924d986c3f60cf/cli_tools/common/utils/daisyutils/daisy_worker.go#L47
> 
> 
> (we would need to confirm that environment variables are propagated properly here).
> Following this approach , we make sure it happens for every workflow (new or old or any).

Thanks for catching this, I updating this part by just calling ParseWorkflow function to create WF in the inspection, please let me know if you've comments.

**Review by darthmelonder — APPROVED** [review]
Ideally, I think this whole "ParseWorkflow" function could be modified to be supported better. First, not scalable parameters, second, it has to rely on being called before the worker runs the workflow.

On other hand, using hooks help us invoking it before workflow seemlessly without requiring to call it explicitly. Environment variable is being populated automatically and makes it more scalable. For now, we can consider this as a follow up as it is more like a long cleanup task.

**Comment by MahmoudNada0** [comment]
> Ideally, I think this whole "ParseWorkflow" function could be modified to be supported better. First, not scalable parameters, second, it has to rely on being called before the worker runs the workflow.
> 
> On other hand, using hooks help us invoking it before workflow seemlessly without requiring to call it explicitly. Environment variable is being populated automatically and makes it more scalable. For now, we can consider this as a follow up as it is more like a long cleanup task.

Hey Jatin,
I moved the clients overriding logic to workflow-hooks in the same PR. please check it and let me know if you've any comments.

**Review by darthmelonder — APPROVED** [review]
Looks good!

**Pull request merged**